common/lib/xmodule/xmodule/video_module/video_handlers.py

```diff
@@ -297,69 +306,46 @@ def transcript(self, request, dispatch):
                 response = self.get_static_transcript(request, transcripts)
                 if response.status_code == 404 and feature_enabled:
-                    # Try to get transcript from edx-val as a last resort.
-                    transcript = get_video_transcript_content(self.edx_video_id, self.transcript_language)
-                    if transcript:
-                        transcript_conversion_props = dict(transcript, output_format=Transcript.SJSON)
-                        transcript = convert_video_transcript(**transcript_conversion_props)
-                        response = Response(
-                            transcript['content'],
-                            headerlist=[('Content-Language', self.transcript_language)],
-                            charset='utf8',
-                        )
-                        response.content_type = Transcript.mime_types[Transcript.SJSON]
-
-                return response
+                    content, filename, mimetype = get_transcript_from_val(
+                        self.edx_video_id,
+                        lang=self.transcript_language,
+                        output_format=Transcript.SJSON
+                    )
+                else:
+                    return response
             except (UnicodeDecodeError, TranscriptsGenerationException) as ex:
                 log.info(six.text_type(ex))
                 response = Response(status=404)
-            else:
-                response = Response(transcript, headerlist=[('Content-Language', language)])
-                response.content_type = Transcript.mime_types['sjson']
+
+            response = Response(
+                content,
+                headerlist=[
+                    ('Content-Disposition', 'attachment; filename="{}"'.format(filename.encode('utf-8'))),
+                    ('Content-Language', self.transcript_language),
+                ],
+                charset='utf8',
+            )
+            response.content_type = mimetype
 
         elif dispatch == 'download':
             lang = request.GET.get('lang', None)
+
             try:
-                transcript_content, transcript_filename, transcript_mime_type = self.get_transcript(
-                    transcripts, transcript_format=self.transcript_download_format, lang=lang
+                content, filename, mimetype = get_transcript(
+                    self.course_id, block_id=self.location.block_id, lang=lang
                 )
-            except (KeyError, UnicodeDecodeError):
+            except (NotFoundError, UnicodeDecodeError):
                 return Response(status=404)
-            except (ValueError, NotFoundError):
-                response = Response(status=404)
-                # Check for transcripts in edx-val as a last resort if corresponding feature is enabled.
-                if feature_enabled:
-                    # Make sure the language is set.
-                    if not lang:
```

Comment:
@muhammad-ammar if lang is not set we will assume a default language, this check should be preserved

Reply:
And I think this is contentstore interface level change :-/